src/rules/noVoidExpressionRule.ts

```diff
@@ -20,13 +20,25 @@ import * as ts from "typescript";
 import * as Lint from "../index";
 import { isTypeFlagSet } from "../language/utils";
 
+const OPTION_ALLOW_ARROW_FUNCTION_SHORTHAND = "allow-arrow-function-shorthand";
```

Comment:
On second thought, can we name this "ignore-arrow-function-shorthand"? I know that we are inconsistent across the codebase in usage of the "allow" and "ignore" prefixes for rule option names (see https://github.com/palantir/tslint/issues/1371), but in this case it does seem more correct to say "ignore linting of arrow function shorthand syntax" because "allow-arrow-function-shorthand" suggests that the rule is checking for a collection of errors, one of which is arrow function shorthand syntax, and this rule option is lenient about that error &mdash; this is not the case; the shorthand syntax is not the error (rather, the implicit non-void return is).

An alternative name that makes sense would be `ignore-implicit-return` (it happens to be that arrow function shorthand is the only way to do that in TS)